### PR TITLE
Add configureActionButtons to the admin interfaces

### PR DESCRIPTION
**Pull request by core23**
Refs https://github.com/sonata-project/SonataAdminBundle/pull/3559

### CODE REVIEW

**Comment by OskarStark** [comment]
I would remove the `[3.0]` Prefix in the title of this PR @core23 and we need to move the Milestone from `3.0` to `4.0`, am i right?

**Comment by Soullivaneuh** [comment]
Please fill your PR of missing parts indicated by the "Needs" labels.

Feel free to ask if you want some precision of what is needed.

**Comment by core23** [comment]
Added the missing needs @Soullivaneuh

File: UPGRADE-4.0.md [hunk 1]
```diff
@@ -1,2 +1,10 @@
 UPGRADE FROM 3.x to 4.0
 =======================
+
+## Admin
+If you have implemented a custom admin, you must adapt the signature of the following new methods to match the one in `AdminInterface` again:
+ * configureActionButtons
```
Comment (Soullivaneuh) [inline UPGRADE-4.0.md]: `configureActionButtons`

File: UPGRADE-4.0.md [hunk 2]
```diff
@@ -1,2 +1,10 @@
...
+
+## AdminExtension
+If you have implemented a custom admin extension, you must adapt the signature of the following new methods to match the one in `AdminExtensionInterface` again:
+ * configureActionButtons
```
Comment (Soullivaneuh) [inline UPGRADE-4.0.md]: `configureActionButtons`

File: Admin/AdminExtensionInterface.php [hunk 3]
```diff
@@ -182,16 +182,15 @@ public function preRemove(AdminInterface $admin, $object);
...
-     */
-    // TODO: Uncomment in next major release
-    // public function configureActionButtons(AdminInterface $admin, $list, $action, $object);
+     /**
+      * Get all action buttons for an action.
+      *
+      * @param AdminInterface $admin
+      * @param array          $list
+      * @param string         $action
+      * @param mixed          $object
```
Comment (Soullivaneuh) [inline Admin/AdminExtensionInterface.php]: Should not be `object`?

**Comment by Soullivaneuh** [comment]
@core23 Do we have doc on 3.x about these methods? If not, could you please add some on another PR?

**Comment by core23** [comment]
Refs https://github.com/sonata-project/SonataAdminBundle/pull/3790

**Comment by OskarStark** [comment]
LGTM

**Comment by Soullivaneuh** [comment]
Thanks @core23.

**Pull request merged**